tests/utils/managed_utils/client.rs

```diff
@@ -108,18 +123,26 @@ impl<'managed_group_lifetime> Client<'managed_group_lifetime> {
         Ok(())
     }
 
-    pub fn receive_messages_for_group(
-        &self,
-        group_id: &GroupId,
-        messages: Vec<MLSMessage>,
-    ) -> Result<(), ClientError> {
+    /// Have the client process the given messages. Returns an error if an error
+    /// occurs during message processing or if no group exists for one of the
+    /// messages.
+    pub fn receive_messages_for_group(&self, messages: &[MLSMessage]) -> Result<(), ClientError> {
         let mut group_states = self.groups.borrow_mut();
-        let group_state = group_states
-            .get_mut(group_id)
-            .ok_or(ClientError::NoMatchingGroup)?;
-        Ok(group_state.process_messages(messages)?)
+        for message in messages {
+            let group_id = GroupId {
+                value: message.group_id(),
```

Comment:
This will not be exposed in future (when we finally cleaned up the API). You can use `GroupId::from_slice`. Or you make your `self.groups` use the raw byte vector instead of the `GroupId`.